fili-core/src/main/java/com/yahoo/bard/webservice/web/DataApiRequest.java

```diff
@@ -2,1088 +2,216 @@
-    }
-
-    public Map<Dimension, Set<ApiFilter>> getApiFilters() {
-        return this.apiFilters;
-    }
-
-    /**
-     * Returns this request's Api filters.
-     *
-     * @return a Map of Dimensions to the set of Api Filters
-     *
-     * @deprecated this method has been deprecated because its name is unclear. Please use {@link #getApiFilters()}.
-     */
-    @Deprecated
-    public Map<Dimension, Set<ApiFilter>> getFilters() {
-        return getApiFilters();
-    }
-
-    /**
-    * Builds and returns the Druid filters from this request's {@link ApiFilter}s.
-    * <p>
-    * The Druid filters are built (an expensive operation) every time this method is called. Use it judiciously.
-    *
-    * @return the Druid filter
-    */
-    public Filter getDruidFilter() {
-        try (TimedPhase timer = RequestLog.startTiming("BuildingDruidFilter")) {
-            return filterBuilder.buildFilters(this.apiFilters);
-        } catch (DimensionRowNotFoundException e) {
-            LOG.debug(e.getMessage());
-            throw new BadApiRequestException(e);
-        }
-    }
-
-    /**
-     * Builds and returns the Druid filters from this request's {@link ApiFilter}s.
-     * <p>
-     * The Druid filters are built (an expensive operation) every time this method is called. Use it judiciously.
-     *
-     * @return the Druid filter
-     *
-     * @deprecated this method has been deprecated because its name is unclear. Please use {@link #getDruidFilter()}.
-     */
-    @Deprecated
-    public Filter getFilter() {
-        return getDruidFilter();
-    }
+    DataApiRequestImpl withFilters(Map<Dimension, Set<ApiFilter>> filters);
 
-    public Map<LogicalMetric, Set<ApiHaving>> getHavings() {
-        return this.havings;
-    }
+    DataApiRequestImpl withHavings(Map<LogicalMetric, Set<ApiHaving>> havings);
 
-    public Having getHaving() {
-        return this.having;
-    }
+    DataApiRequestImpl withHaving(Having having);
 
-    public LinkedHashSet<OrderByColumn> getSorts() {
-        return this.sorts;
-    }
+    DataApiRequestImpl withSorts(LinkedHashSet<OrderByColumn> sorts);
 
-    public OptionalInt getCount() {
-        return count == 0 ? OptionalInt.empty() : OptionalInt.of(count);
-    }
+    DataApiRequestImpl withCount(int count);
 
-    public OptionalInt getTopN() {
-        return topN == 0 ? OptionalInt.empty() : OptionalInt.of(topN);
-    }
+    DataApiRequestImpl withTopN(int topN);
 
-    public DateTimeZone getTimeZone() {
-        return timeZone;
-    }
+    DataApiRequestImpl withAsyncAfter(long asyncAfter);
 
-    public DruidFilterBuilder getFilterBuilder() {
-        return filterBuilder;
-    }
+    DataApiRequestImpl withTimeZone(DateTimeZone timeZone);
 
-    public Optional<OrderByColumn> getDateTimeSort() {
-        return dateTimeSort;
-    }
+    DataApiRequestImpl withFilterBuilder(DruidFilterBuilder filterBuilder);
+    DataApiRequestImpl withHavingApiGenerator(HavingGenerator havingApiGenerator);
```

Comment:
Should this method exist?  Is it part of the interface?  Is there a reasonable time it would be called?